### PR TITLE
[FAL-229] Docker devstack for cert-manager

### PR DESCRIPTION
**Pull request by ChrisChV**
Simple devstack added to run the unit tests easily

**Jira tciket:** [FAL-229](https://tasks.opencraft.com/browse/FAL-229)

- New docker-compose.yml file added that creates one container running the consul service and another with python3.6
- New Makefile.yml file added with commands to start/stop/destroy the devstack and one command to run the unit tests on the python3.6 container
- README.md updated with this new way to run tests

**How to test**

- Read and follow the new instructions on README

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+start: ## Start the devstack
+	docker-compose -f docker-compose.yml up -d
```
Comment (swalladge) [inline Makefile:2]: @ChrisChV `-f docker-compose.yml` can be omitted, because this is the default value.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+stop: ## Stop the devstack
+	docker-compose -f docker-compose.yml stop
+
+destroy: ## Remove all containers and volumes
+	docker-compose -f docker-compose.yml down -v
+
+test: ## Run tests
+	docker exec -e COLUMNS="`tput cols`" -e LINES="`tput lines`" -it cert-manager.devstack.test /bin/bash /opt/scripts/run_tests.sh
```
Comment (swalladge) [inline Makefile:11]: Please configure your editor to properly add the final newline at the end of files. :)

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    environment: 
+      CONSUL_BIND_INTERFACE: eth0
+      PIPENV_VENV_IN_PROJECT: "true"
+      LC_ALL: "C.UTF-8"
+      LANG: "C.UTF-8"
+    tty: true
+    ports:
+      - "8500:8500"
+    volumes: 
+      - "$PWD:/opt/"
```
Comment (swalladge) [inline docker-compose.yml:15]: Check for trailing whitespace. ;)

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    volumes: 
+      - "$PWD:/opt/"    
+    networks:
+      cert-manager_default:
+        ipv4_address: 192.168.176.3
+  test:
+    container_name:  cert-manager.devstack.test
+    image: python:3.6
+    environment: 
+      CONSUL_HTTP_ADDR: 192.168.176.3:8500
```
Comment (swalladge) [inline docker-compose.yml:23]: The custom networks can be removed, and this value replaced with `cert-manager.devstack.consul:8500` (relying on docker's pseudo dns system).

**Review by swalladge — CHANGES_REQUESTED** [review]
@ChrisChV looks good, and works perfectly!  I left some requests mainly about style changes. :)

File: README.md [hunk 5]
```diff
@@ -8,6 +8,20 @@ This repository contains a certificate manager based on Certbot, using the Consu
 Running the unit tests
 ----------------------
 
+There are two options to run the unit tests:
+
+### Devstack way:
+
+To run the unit tests in a devstack (containers), you need to start the devstack:
```
Comment (mtyaka) [inline README.md:15]: Can we replace "containers" with "docker" just to make it more obvious that you need to have docker installed for this to work?

File: scripts/run_tests.sh [hunk 6]
```diff
@@ -0,0 +1,4 @@
+pip install --user pipenv
+cd /opt
+python -m pipenv install --dev --ignore-pipfile
+python -m pipenv run pytest
```
Comment (mtyaka) [inline scripts/run_tests.sh:4]: Nit: no newline at end of file

**Comment by mtyaka** [comment]
Nice work! Please add the missing newline, but then you can go ahead and merge :+1:

- [x] I tested this: I verified that the make commands work perfectly.
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~ n/a
- [x] Includes documentation

**Review by swalladge — APPROVED** [review]

**Pull request merged**